src/main/java/com/epam/training/gen/ai/service/KernelHistoryService.java

```diff
@@ -0,0 +1,81 @@
+package com.epam.training.gen.ai.service;
+
+import com.epam.training.gen.ai.exception.GenAiBadRequestException;
+import com.microsoft.semantickernel.Kernel;
+import com.microsoft.semantickernel.semanticfunctions.KernelFunction;
+import com.microsoft.semantickernel.semanticfunctions.KernelFunctionArguments;
+import com.microsoft.semantickernel.services.chatcompletion.ChatHistory;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service class for interacting with the AI kernel, maintaining chat history.
+ * <p>
+ * This service provides a method to process user prompts while preserving chat history.
+ * It uses the {@link Kernel} to invoke AI responses based on the user's input and the
+ * previous chat context. The conversation history is updated after each interaction.
+ */
+@Slf4j
+@Service
+public class KernelHistoryService {
+  private final Kernel kernel;
+  private final ChatHistory chatHistory = new ChatHistory();
```

Comment:
You could use spring bean with "session" scope and proxy mode "target class", so chat history will be in session scope of a specific client
```java
    @Bean
    @Scope(scopeName = "session", proxyMode = ScopedProxyMode.TARGET_CLASS)
    public ChatHistory chatHistory() {
        return new ChatHistory();
    }
```

Reply:
I thought about it, but it won't work. As we know REST is stateless and when connecting a REST client to the application we won't have a session.
Anyway, Thank you for your idea!)

Reply:
Thank you for your idea.